新規投稿機能(Posts.create)の作成

tattu501 opened the pull request:
#何か
新規投稿のアクション、およびビューを追加

#変更ファイル
posts.controller.rb
posts/new.html.erb
posts/shared/_error_messages.html.erb
post.rb
Gemfile
Gemfile.lock
config/application.rb
config/locales/ja.yml

msgyu committed:
RUN "rails g controller posts"

msgyu committed:
Add index in PostsController

indexのviewを作成するために、コントローラーを作成した

msgyu committed:
Add resources posts in routes.rb

indexのview作成のために、追加。
CRUD機能すべてを実装するため、resourcesで記述している。

msgyu committed:
Create index.html.erb of posts

msgyu committed:
Add footer

msgyu committed:
部分テンプレートを読み込み

こちらにfooterを記述する

msgyu committed:
footerを追加

msgyu committed:
Create footer.scss

msgyu committed:
部分テンプレートのパスを変更

sharedフォルダに_footer.html.erbを入れた

msgyu committed:
Edit footer

msgyu committed:
change class in footer

msgyu committed:
Add css in footer.scss

msgyu committed:
階層を変更

msgyu committed:
Add css of a in reset.scss

aタグのリセットcssが記述されていないため、追記した。

msgyu committed:
change color of h4  in footer

tattu501 committed:
routes.rbにresources :postsを追加

ヘッダのビューを確認するため、postsのルーティングを追加

msgyu committed:
Add  column of post model

tattu501 committed:
application.scssにheader.scssのインポートを追加

ヘッダの各ボタンにCSSを反映するため

tattu501 committed:
header.scssにヘッダのSCSSを実装

ヘッダの各ボタンに関して、配置や色を指定するため。

tattu501 committed:
posts.controller.rbにindexアクションを追加

ヘッダの実装状況を確認するため

tattu501 committed:
application.html.erbにheaderの部分テンプレート呼び出しを実装

ヘッダのビューを呼び出すため、posts/header.html.erbをrenderする処理を追加

tattu501 committed:
views/postsにindex.html.erbを追加

ヘッダのビューを確認するため

tattu501 committed:
RUN "rails g controller posts"

ヘッダのビューを確認するためにpostsコントローラを作成した際に出来たファイル

tattu501 committed:
_header.html.erbを作成

ヘッダ用のhtmlを作成するため

ozeryu7 committed:
users_controllerを追加

ozeryu7 committed:
user_controllerを追加した時にできたファイル4つを追加

msgyu committed:
Add Post model

msgyu committed:
Merge branch 'create__model__post' into posts.index

msgyu committed:
define @posts

msgyu committed:
Move index.html.erb

msgyu committed:
Add root in routes.rb

msgyu committed:
Add each in index.html.erb

Display the post repeatedly

msgyu committed:
Add a relation of post and user

msgyu committed:
Merge branch 'create__model__post' into posts.index

msgyu committed:
変数でPostを表示 in index.html.erb

msgyu committed:
投稿順を日付順でDESC

msgyu committed:
Add posts.scss

msgyu committed:
imposrt posts.scss

msgyu committed:
Add includes(:user) to @posts

N+1問題を発生させないために追加

msgyu committed:
投稿者の場合、編集・削除の選択肢を表示

# 何か
投稿者本人の場合、編集・削除ボタンを表示させる。
そんために、post.user_idとcurrent_user.id(ログインuserのid)が一致しているかif文で判定している。

msgyu committed:
Add before_action :authenticate_user! in posts_controller

msgyu committed:
rootの位置を変更

rootを位置に配置した。これでより確認しやすくなった。

msgyu committed:
Merge branch 'header' into posts.index

msgyu committed:
Add link_to

msgyu committed:
Change a to li

msgyu committed:
User認証機能が未完成なため、foreign_keyをコメントアウト

Userの新規登録機能とログイン機能実装後に、コメントアウトを解除する。
migrateしている場合は、rake db:resetを実施してコメントアウト を反映させてから開発します。

tattu501 committed:
headerの修正

headerの余計なコメント（ルートパスへのリンクを設置）を削除、ならびにIF文でログインしているかで表示内容を分岐するように修正しました。

tattu501 committed:
postモデルからuser_idを削除

postモデルのマイグレーションファイルからforeign_keyをコメントアウトされたのを反映するため、rake db:rollbackのあとrake db:migrateを実施

tattu501 committed:
posts_controller.rbにnewアクションを追加

postの登録機能を実装するため、newアクションを追加した

tattu501 committed:
posts/new.html.erbを作成

postの新規投稿ページを実装するため、ビューファイルを作成
※中身はありません。空ファイルを作成

tattu501 committed:
posts/new.html.erbに投稿フォームを実装

postの投稿機能を追加するため、ビューファイルに投稿フォームを実装した

tattu501 committed:
posts.controller.rbからbefore_actionのauthenticate_user!をコメントアウト

新規投稿ページの確認のため、before_action :authenticate_user!, except: [:index, :show]をコメントアウト

tattu501 committed:
posts/shared/_error_messages.html.erbを作成

エラーメッセージ用のパーシャルファイルを作成
※中身はありません

tattu501 committed:
posts/new.html.erbを作成

エラーメッセージ用のパーシャルを除き、新規投稿ページのビューが完成した。

tattu501 committed:
posts/shared/_error_messages.html.erbの中身を作成

新規投稿ページのエラーメッセージを作成

tattu501 committed:
gem 'rails-i18n'をインストール

バリデーションのメッセージを日本語化するため、gem 'rails-i18n'を
Gemfile.lockに追記してbundle installを実施

tattu501 committed:
config/application.rbにエラーメッセージ日本語化用の定義を追加

新規投稿のバリデーションエラー時、メッセージが日本語化されるようにするため、config/application.rbに定義を追加

tattu501 committed:
post.rbに投稿内容のバリデーションを追加

投稿内容がブランク、もしくは140字を超える場合にエラーとするよう、バリデーションを追加した。
なお、belongs_to :userがあると、ユーザ登録機能が未完成のため"userを入れてください。"と出る。バリデーションを正確に検証するため、belongs_to :userをコメントアウトした。

tattu501 committed:
posts.controller.rbにcreateアクションの中身を実装

新規投稿の機能を作成するため、新規投稿用のcreateアクションの内容を実装した。
※前回コミット時にはcreateアクションのガワだけ作成

tattu501 committed:
/posts/shared/_error_messages.html.erbを修正

不要な箇所を削除し、バリデーションエラーのメッセージ実装を完了しました。

tattu501 committed:
config/locales/ja.ymlを追加

バリデーションエラーを日本語するため、ja.ymlを定義した。

ozeryu7 committed:
application.htmlのヘッダーにif user_signed_in?の条件分岐を追加

ヘッダーにサインインしているかで条件分岐を設定

ozeryu7 committed:
application.controllerにストロングパラメーター追加

ストロングパラメーターを追加

ozeryu7 committed:
userモデルにバリデーションを追加

名前、メール、パスワードのバリデーションを追加

ozeryu7 committed:
postテーブル追加時schemaに自動追記

ozeryu7 committed:
config.routesにresourcesを追加

routesにroot to:indexとresources:postsを追加

msgyu committed:
RUN "rails g controller posts"

画面表示するために、土台となるposts_controllerを作成

msgyu committed:
Merge branch 'header' into origin/user.devise

msgyu committed:
Add header

msgyu committed:
partsフォルダに_header.html.erbを移動

_header.html.erbは共通で利用するため、userのログイン時に表示させる必要がある。そのため、共通で呼び出しできる場所に移行した

msgyu committed:
user名をheaderに表示

msgyu committed:
Merge branch 'origin/user.devise' into posts.create

msgyu committed:
_header.html.erbをsharedフォルダに移行

footerが格納されているsharedにheader.html.erbを入れ、共通フォルダに入れた

msgyu committed:
RUN"rails db:migrate:reset"

userモデルのforeign_keyを追加した

msgyu committed:
before_action :authenticate_user!のコメントアウトを解除

Userモデルを追加したので、 before_action :authenticate_user!が利用できるようになった。そのためにコメントアウト を解除した

msgyu committed:
コメント数を数字で表示

tattu501 committed:
Merge branch 'origin/user.devise' into posts.create

msgyu committed:
コメント・いいね数を数字で表示

仮置きとして１を記述した

tattu501 committed:
post.rbのbelongs_to :userを再実装

belongs_to :userユーザのコメントアウトを外し、userモデルとpostモデルを紐付ける

tattu501 committed:
postモデルに外部キーuser_idを再実装

rake db:resetを行い、外部キーuser_idを再度実装した

tattu501 committed:
posts.controller.rbにbefore_actionを再実装

コメントアウトしていたbefore_action :authenticate_user!, except: [:index, :show]を再度実装した

tattu501 committed:
_header.html.erbの「投稿する」ボタンにリンクを付与

「投稿する」ボタンに新規投稿ページに遷移するリンクを付与した。

msgyu committed:
Add gem 'pry-rails'

msgyu committed:
@postsのincludes(:user)を先に記述

先に呼び出すことで処理速度を向上させるため

msgyu committed:
= を追加して投稿情報を表示

msgyu committed:
=を追加して投稿情報を表示

msgyu committed:
postをlink_toの中に格納

これでpostのdivをクリックするだけでposts.showに移行する。

msgyu committed:
Merge branch 'posts.index' into posts.create

tattu501 committed:
postモデルの外部キーuser_idを再実装

user認証が完成したので、postモデルに元々あった外部キーuser_idを再度実装した

tattu501 committed:
post.controller.rbのcreateアクションをuserモデルと紐付けた

user認証が完成したので、Post.newからcurrent_user.posts.buildに変更し、postの外部キーuser_idにユーザのIDが入るようにした。

tattu501 committed:
posts/index.html.erbに投稿内容を表示できるようにした

新規投稿機能が完成したので、投稿した内容が一覧ページに表示されるようにした。

tattu501 committed:
pullした内容を反映

pullした最新の内容を反映

tattu501 committed:
pullした内容を反映

pullした最新の内容を反映

tattu501 committed:
application.html.erbの不要な箇所を削除

applicaiton.html.erbに不要な文字列が残っていたので削除した

msgyu committed:
ログアウト時のエラーを防ぐため、if user_signed_in?を追加

ログアウト時はcurrent_userの情報を取得できないため、エラーが発生する。そのため、if user_signed_in?でログインしているか判定している

msgyu committed:
Merge branch 'posts.index' into posts.create

msgyu committed:
Revert "pullした内容を反映"

This reverts commit 30c51af9a7aa13072bf1c9f0034ec3ff95876ee5.

msgyu committed:
チームメンバーのpullを修正

msgyu committed:
チームメンバーのpullを修正

msgyu committed:
チームメンバーのpullを修正

msgyu committed:
Revert "チームメンバーのpullを修正"

This reverts commit f77ab5a174f5306701ad8c203bdbeabcad327cc7.

msgyu committed:
Revert "チームメンバーのpullを修正"

This reverts commit 67e59edafbd8d0a6db44411b83754e7fc3883674.

msgyu committed:
Revert "チームメンバーのpullを修正"

This reverts commit e22044c2977da92e6b4f2ac536397a49e93f8335.

msgyu committed:
Revert "Revert "pullした内容を反映""

This reverts commit 63fdcfeb04cdec41e747d3dda9df623f5a9e16f0.

msgyu commented on app/assets/javascripts/posts.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
posts.coffeeファイルの削除

tattu501 replied:
申し訳ございません、コメントを失念しておりました。上記ファイルを削除しました。

msgyu commented on app/assets/javascripts/users.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
user.coffeeは使用しないので削除しましょう

tattu501 replied:
user.coffeeを削除しました。

msgyu commented on app/assets/stylesheets/users.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the users controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
users.scssを削除しておきましょう

tattu501 replied:
users.scssを削除しました。

msgyu commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,11 @@
+class UsersController < ApplicationController
+  def index
+    @users = User.all
+  end
+
+  def new
+    @user = User.new
+  end
+
+  
+end
```
users_controller.rbの削除
今回は使用しません

tattu501 replied:
users_controller.rbを削除しました。

msgyu commented on app/assets/stylesheets/posts.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the posts controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
使用していないので、posts.scssを削除しておきましょう

tattu501 replied:
posts.scssを削除しました。

msgyu commented on app/views/posts/_header.html.erb:
```diff
@@ -0,0 +1,28 @@
+<header class="header">
+  <div class="header__bar">
+    <h1>
+      <%= link_to "転クエ掲示板", root_path %>
+    </h1>
+    <%# ログイン状態に応じて表記を変更すること %>
+    <ul class="header__bar__user_nav">
+      <li class="user_name">
+        <span>ゲストユーザ さん</span>
+      </li>
+      <li class="header__bar__user_nav__btn">
+        <%= link_to "ログアウト", "#" %>
+      </li>
+      <li class="header__bar__user_nav__btn">
+        <%= link_to "投稿する", "#" %>
+      </li>
+      <li class="header__bar__user_nav__btn">
+        <%= link_to "投稿する", "#" %>    
+      </li>
+      <li class="header__bar__user_nav__btn">
+        <%= link_to "ログイン", "#" %> 
+      </li>
+      <li class="header__bar__user_nav__btn">
+        <%= link_to "新規登録", "#" %> 
+      </li>
+    </ul>
+  </div>
+</header>
```
このファイル削除しましょう。
partsフォルダに格納されている_header.html.erbは、sharedフォルダに移行完了済みです。
なので、partsフォルダに入っているheader.html.erbは古いのでこれを削除してください。

tattu501 replied:
上記ファイルを削除しました。

msgyu commented on db/migrate/20201030162743_add_user_id_to_posts.rb:
```diff
@@ -0,0 +1,4 @@
+class AddUserIdToPosts < ActiveRecord::Migration[5.2]
+  def change
+  end
+end
```
不要なマイグレーションファイルなので、削除しましょう
削除完了後、rake db:reset を実施して、schemaを更新してください

tattu501 replied:
上記のマイグレーションファイルを削除しました。

msgyu commented on db/schema.rb:
```diff
@@ -10,11 +10,19 @@
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2020_08_08_145604) do
+ActiveRecord::Schema.define(version: 2020_10_30_162743) do
```
使用していないmigrationファイルの日付になっているので、migrationファイル削除後、rake db:reset で更新し、日付を修正してください

tattu501 replied:
申し訳ございません。マイグレーションファイルを削除後、rake db:reset をしましたが、日付が変わりませんでした。

tattu501 committed:
不要なファイルを削除

不要なファイルを削除した。
・app/assets/javascripts/posts.coffee
・app/assets/javascripts/users.coffee
・app/assets/stylesheets/users.scss
・app/controllers/users_controller.rb
・app/assets/stylesheets/posts.scss
・app/views/posts/_header.html.erb
・db/migrate/20201030162743_add_user_id_to_posts.rb

msgyu commented:
LGTM

masato-sa93 reviewed (approved):
LGTMですー！
おそらく別のプルリクをマージするとコンフリクト起こすので、そこを注意してください！
